### DISCUSSION TITLE
if we build GLSP solution, is it possible to run with that in local non web environment

### DISCUSSION THREAD

**Discussion by midhun8454** [eclipse-glsp/glsp#434] [General] [answered]
Means if i install a plugin to eclipse or vscode will it work ? or run in my laptop ?

**Comment by planger** [comment, marked as answer]
Yes, you can deploy a GLSP diagram editor as VSCode extension or Theia extension or Eclipse plugin that runs locally.
VSCode: https://github.com/eclipse-glsp/glsp-vscode-integration
Theia: Theia can be dejployed as electron app (local) or cloud IDE. The GLSP extension is the same: https://github.com/eclipse-glsp/glsp-theia-integration
Eclipse: https://github.com/eclipse-glsp/glsp-eclipse-integration

**Reply by midhun8454** [reply]
is all there a node dependency , as i can see all works on node version ~12 and not more than 13  ?

**Reply by planger** [reply]
Yes the node version is mostly dictated by Theia to which we want to be compatible.
So we use the same version range as Theia.
https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites

**Reply by midhun8454** [reply]
@planger thanks a lot for the info.. one last question is Theia prod based code or its still in beta version ? as i can see via vscode its evolving , not 100% and also same with traditional eclipse? so do we need to go with standalone application ?

**Reply by planger** [reply]
Yes, Theia is production ready and already used for many production use cases today (see https://blogs.eclipse.org/post/brian-king/eclipse-theia-builds-momentum).
